components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1385,11 +1386,36 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
 
     public async getTeams(): Promise<Team[]> {
         // Note: this operation is per-user only, hence needs no resource guard
-        const user = this.checkUser("deleteEnvVar");
-
+        const user = this.checkUser("getTeams");
         return this.teamDB.findTeamsByUser(user.id);
     }
 
+    public async getTeamMembers(teamId: string): Promise<TeamMemberInfo[]> {
+        this.checkUser("getTeamMemberships");
+        const team = await this.teamDB.findTeamById(teamId);
+        if (!team) {
+            throw new ResponseError(ErrorCodes.NOT_FOUND, "Team not found");
+        }
+        const memberships = await this.teamDB.findMembershipsByTeam(team.id);
+        await this.guardAccess({ kind: "team", subject: team, memberships }, "get");
+        return Promise.all(memberships.map(async (m: DBTeamMembership): Promise<TeamMemberInfo> => {
+            const member = await this.userDB.findUserById(m.userId);
```

Comment:
Depending on the team size this can become expensive. A single SQL query would be better.

Reply:
https://github.com/gitpod-io/gitpod/issues/4470